### ISSUE TITLE
Crash Bug Involving Contract Balances

### ISSUE THREAD

**Issue by patrickdugan** [BlockPo/BlockPo-to-Tradelayer#486]
This is a stub for a hard to replicate little bugger. We had an error-message-free crash July 25th log output is here:

tl;dr on this output, setting up an oracle, commits, trade 1 contract, no problem, do it again for 90, the log is a little twisted at the end with

2022-07-25T11:39:11Z Instant_x_Trade: old first position: -1, new first position: -91 
2022-07-25T11:39:11Z Instant_x_Trade: old second position: 1, new second position: 91 
2022-07-25T11:39:11Z Instant_x_Trade: Status_marker0: ShortPosIncreased, Status_taker0: LongPosIncreased 
2022-07-25T11:39:11Z Instant_x_Trade: amount_forsale: 90
2022-07-25T11:39:11Z Instant_x_Trade: newPosMaker: 0
2022-07-25T11:39:11Z Instant_x_Trade: newPosTaker: 0

Ends abruptly with:

2022-07-25T11:39:11Z logicMath_Contract_Instant(): End of Logic Instant Contract Trade

Restart ran into problems syncing until Val deleted the wallet folder (with no back-up so we can't reimport the keys to check if they are somehow corrupted in other nodes).

Need to spam test to reproduce.

2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430946, 2022-07-25 idx= 2); txid: 2cfa694ffedd22e18195621ace11ae9823c886e01a9d3b9e71526098849cebc9
2022-07-25T11:38:53Z parseTransaction(): strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, spReference: , strReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, single_pkt: 001d6f7261636c6538000a80f28ba8090201, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 29
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 29 (Future Contract)
2022-07-25T11:38:53Z 	 leverage: 1
2022-07-25T11:38:53Z 	 messageType: 29
2022-07-25T11:38:53Z 	 contractName: oracle8
2022-07-25T11:38:53Z 	 amount of contracts : 10
2022-07-25T11:38:53Z 	 effective price : 2500000000
2022-07-25T11:38:53Z 	 trading action : 2
2022-07-25T11:38:53Z logicMath_ContractDexTrade(): Leverage in register: 0, position: 0 
2022-07-25T11:38:53Z logicMath_ContractDexTrade(): setting leverage: 1 
2022-07-25T11:38:53Z updateRecord(): now64: 0
2022-07-25T11:38:53Z checkContractReserve(): amountToReserve: 200025000
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(2cfa694ffedd22e18195621ace11ae9823c886e01a9d3b9e71526098849cebc9, valid=YES, reason=0, block= 2430946, type= 29, value= 0)
2022-07-25T11:38:53Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430947, 2022-07-25 idx= 1); txid: 6abb6a3179d8662c33c419b88edf9b5b0c52d107ea9c095a5c500185b7f387a0
2022-07-25T11:38:53Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, single_pkt: 0069088080bb8b939b448080d287e2bc2d80c0c6c9faeb38, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 105
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 105 (Oracle Set Address)
2022-07-25T11:38:53Z 	 version: 0
2022-07-25T11:38:53Z 	 oracle high price: 300000000000000
2022-07-25T11:38:53Z 	 oracle low price: 200000000000000
2022-07-25T11:38:53Z 	 oracle close price: 250000000000000
2022-07-25T11:38:53Z 	 contractId: 8
2022-07-25T11:38:53Z updateCD(): updated entry for CD 8 successfully
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(6abb6a3179d8662c33c419b88edf9b5b0c52d107ea9c095a5c500185b7f387a0, valid=YES, reason=0, block= 2430947, type= 105, value= 0)
2022-07-25T11:38:53Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430950, 2022-07-25 idx= 2); txid: de2675bf49a841d1e87f45e733b1b9b50df0b937058233e302d477016221903f
2022-07-25T11:38:53Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, single_pkt: 0032020074657374546f6b656e390075726c54657374546f6b656e39006461746154657374546f6b656e390080d0dbc3f40200, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 50
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 50 (Create Property - Fixed)
2022-07-25T11:38:53Z 	   property type: 2 (divisible)
2022-07-25T11:38:53Z 	prev property id: 0
2022-07-25T11:38:53Z 	            name: testToken9
2022-07-25T11:38:53Z 	             url: urlTestToken9
2022-07-25T11:38:53Z 	            data: dataTestToken9
2022-07-25T11:38:53Z 	           value: 1000.00000000
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(de2675bf49a841d1e87f45e733b1b9b50df0b937058233e302d477016221903f, valid=YES, reason=0, block= 2430950, type= 50, value= 100000000000)
2022-07-25T11:38:53Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430950, 2022-07-25 idx= 3); txid: 8fafb65d5cfff2d2cb894cbd27314e38c9f571f0c76ac0254f039b64ee7e58ff
2022-07-25T11:38:53Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, single_pkt: 00676f7261636c653900c03e80c8afa0050980ade2040100, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 103
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 103 (Create Oracle Contract)
2022-07-25T11:38:53Z 	 version: 0
2022-07-25T11:38:53Z 	 messageType: 103
2022-07-25T11:38:53Z 	 blocks until expiration : 8000
2022-07-25T11:38:53Z 	 notional size : 1410065408
2022-07-25T11:38:53Z 	 collateral currency: 9
2022-07-25T11:38:53Z 	 margin requirement: 10000000
2022-07-25T11:38:53Z 	 name: oracle9
2022-07-25T11:38:53Z 	 oracleAddress: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ
2022-07-25T11:38:53Z 	 backupAddress: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ
2022-07-25T11:38:53Z 	 prop_type: 6
2022-07-25T11:38:53Z 	 inverse quoted: 1
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(8fafb65d5cfff2d2cb894cbd27314e38c9f571f0c76ac0254f039b64ee7e58ff, valid=YES, reason=0, block= 2430950, type= 103, value= 0)
2022-07-25T11:38:53Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430951, 2022-07-25 idx= 1); txid: 463d4ec30368ecaa20b02b20046b734ab06df05afe8224df563e873b14e22ac8
2022-07-25T11:38:53Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, strReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, single_pkt: 00000980c8afa025, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 0
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 0 (Simple Send)
2022-07-25T11:38:53Z 	        property: 9 (SP token: 9)
2022-07-25T11:38:53Z 	           value: 100.00000000
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(463d4ec30368ecaa20b02b20046b734ab06df05afe8224df563e873b14e22ac8, valid=YES, reason=0, block= 2430951, type= 0, value= 10000000000)
2022-07-25T11:38:53Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:38:53Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:38:53Z parseTransaction(block=2430951, 2022-07-25 idx= 2); txid: 77f00471e890b750286ad692b5a6c9a44c9f52c39bc01ca33f6d62673cdec4fa
2022-07-25T11:38:53Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:38:53Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, single_pkt: 00690980bcc1960b80a8d6b90780f28ba809, tlClass: 4 
2022-07-25T11:38:53Z interpret_TransactionType(): type: 105
2022-07-25T11:38:53Z 	------------------------------
2022-07-25T11:38:53Z 	         version: 0, class D
2022-07-25T11:38:53Z 	            type: 105 (Oracle Set Address)
2022-07-25T11:38:53Z 	 version: 0
2022-07-25T11:38:53Z 	 oracle high price: 3000000000
2022-07-25T11:38:53Z 	 oracle low price: 2000000000
2022-07-25T11:38:53Z 	 oracle close price: 2500000000
2022-07-25T11:38:53Z 	 contractId: 9
2022-07-25T11:38:53Z updateCD(): updated entry for CD 9 successfully
2022-07-25T11:38:53Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:38:53Z recordTX(77f00471e890b750286ad692b5a6c9a44c9f52c39bc01ca33f6d62673cdec4fa, valid=YES, reason=0, block= 2430951, type= 105, value= 0)
2022-07-25T11:39:10Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:10Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:10Z parseTransaction(block=2430952, 2022-07-25 idx= 1); txid: b15b530317650517bbb1e14867ace91a550de0e634a09ff96b1487e4cc015e40
2022-07-25T11:39:10Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:39:10Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c098094ebdc03, tlClass: 4 
2022-07-25T11:39:10Z interpret_TransactionType(): type: 108
2022-07-25T11:39:10Z 	------------------------------
2022-07-25T11:39:10Z 	         version: 0, class D
2022-07-25T11:39:10Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:10Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:10Z 	 version: 0
2022-07-25T11:39:10Z 	 propertyId: 9
2022-07-25T11:39:10Z 	 amount commited: 1000000000
2022-07-25T11:39:10Z recordNewChannel(): OK
2022-07-25T11:39:10Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:10Z recordTX(b15b530317650517bbb1e14867ace91a550de0e634a09ff96b1487e4cc015e40, valid=YES, reason=0, block= 2430952, type= 108, value= 0)
2022-07-25T11:39:10Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:10Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:10Z parseTransaction(block=2430952, 2022-07-25 idx= 2); txid: 0eb7d3eccd01995b31d37e31fa2a52f74e2bfabe23909b333a9e8148557a7383
2022-07-25T11:39:10Z parseTransaction(): strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ 
2022-07-25T11:39:10Z parseTransaction(): mp_tx object: strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, spReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c098094ebdc03, tlClass: 4 
2022-07-25T11:39:10Z interpret_TransactionType(): type: 108
2022-07-25T11:39:10Z 	------------------------------
2022-07-25T11:39:10Z 	         version: 0, class D
2022-07-25T11:39:10Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:10Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:10Z 	 version: 0
2022-07-25T11:39:10Z 	 propertyId: 9
2022-07-25T11:39:10Z 	 amount commited: 1000000000
2022-07-25T11:39:10Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:10Z recordTX(0eb7d3eccd01995b31d37e31fa2a52f74e2bfabe23909b333a9e8148557a7383, valid=YES, reason=0, block= 2430952, type= 108, value= 0)
2022-07-25T11:39:10Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:10Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:10Z parseTransaction(block=2430952, 2022-07-25 idx= 3); txid: 9a81ead6270afd8cf941f2fba3c358c171a12965798ceb4bfde22c4b03fe1526
2022-07-25T11:39:10Z parseTransaction(): strSender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb 
2022-07-25T11:39:10Z parseTransaction(): mp_tx object: strSender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, spReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, strReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, single_pkt: 00720901cbb0940180f28ba8090101, tlClass: 4 
2022-07-25T11:39:10Z interpret_TransactionType(): type: 114
2022-07-25T11:39:10Z 	------------------------------
2022-07-25T11:39:10Z 	         version: 0, class D
2022-07-25T11:39:10Z 	            type: 114 (Channel Contract Instant Trade)
2022-07-25T11:39:10Z sinterpret_Contract_Instant): inside interpret function!!!!!
2022-07-25T11:39:10Z sinterpret_Contract_Instant): function returning !!!!!
2022-07-25T11:39:10Z 	 version: 0
2022-07-25T11:39:10Z 	 messageType: 114
2022-07-25T11:39:10Z 	 contractId: 9
2022-07-25T11:39:10Z 	 amount : 1
2022-07-25T11:39:10Z 	 blockfor_expiry : 2431051
2022-07-25T11:39:10Z 	 price : 2500000000
2022-07-25T11:39:10Z 	 trading action : 1
2022-07-25T11:39:10Z 	 leverage : 1
2022-07-25T11:39:10Z logicMath_Contract_Instant: AmountToReserve: 10000000
2022-07-25T11:39:10Z logicMath_Contract_Instant: sender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, channel Address: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:10Z logicMath_Contract_Instant(): first address reserve done
2022-07-25T11:39:10Z logicMath_Contract_Instant(): second address reserve done
2022-07-25T11:39:10Z updateRecord(): now64: 0
2022-07-25T11:39:10Z updateRecord(): now64: 0
2022-07-25T11:39:10Z Instant_x_Trade: old first position: 0, new first position: -1 
2022-07-25T11:39:10Z Instant_x_Trade: old second position: 0, new second position: 1 
2022-07-25T11:39:10Z Instant_x_Trade: Status_marker0: OpenShortPosition, Status_taker0: OpenLongPosition 
2022-07-25T11:39:10Z Instant_x_Trade: amount_forsale: 1
2022-07-25T11:39:10Z Instant_x_Trade: newPosMaker: 0
2022-07-25T11:39:10Z Instant_x_Trade: newPosTaker: 0
2022-07-25T11:39:10Z 
Check here CDEx:	 nBlockNow = 2430952	 twapBlockg = 2430944
2022-07-25T11:39:10Z 
cdextwap_ele.size() = 1
2022-07-25T11:39:10Z logicMath_Contract_Instant(): End of Logic Instant Contract Trade

2022-07-25T11:39:10Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:10Z recordTX(9a81ead6270afd8cf941f2fba3c358c171a12965798ceb4bfde22c4b03fe1526, valid=YES, reason=0, block= 2430952, type= 114, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430953, 2022-07-25 idx= 2); txid: c3d57a2b1d55bbb2f551008c7bea31e496598ed97822b718553631a17cae45c9
2022-07-25T11:39:11Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, single_pkt: 00690980d88ee16f8090dfc04a80f4f6905d, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 105
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 105 (Oracle Set Address)
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 oracle high price: 30000000000
2022-07-25T11:39:11Z 	 oracle low price: 20000000000
2022-07-25T11:39:11Z 	 oracle close price: 25000000000
2022-07-25T11:39:11Z 	 contractId: 9
2022-07-25T11:39:11Z updateCD(): updated entry for CD 9 successfully
2022-07-25T11:39:11Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:11Z recordTX(c3d57a2b1d55bbb2f551008c7bea31e496598ed97822b718553631a17cae45c9, valid=YES, reason=0, block= 2430953, type= 105, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430955, 2022-07-25 idx= 1); txid: ef8a31beda54c2ee8cc48daa4f8b45216aefc9209bced3081336b9a14317cb27
2022-07-25T11:39:11Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: , strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c0980d293ad03, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 108
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:11Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 propertyId: 9
2022-07-25T11:39:11Z 	 amount commited: 900000000
2022-07-25T11:39:11Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:11Z recordTX(ef8a31beda54c2ee8cc48daa4f8b45216aefc9209bced3081336b9a14317cb27, valid=YES, reason=0, block= 2430955, type= 108, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430955, 2022-07-25 idx= 2); txid: 863957dbf25e6386892abbe1ad2baa0e7f54ee5acc743814b5ddd5e747a1b3f8
2022-07-25T11:39:11Z parseTransaction(): strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, spReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c0980d293ad03, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 108
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:11Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 propertyId: 9
2022-07-25T11:39:11Z 	 amount commited: 900000000
2022-07-25T11:39:11Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:11Z recordTX(863957dbf25e6386892abbe1ad2baa0e7f54ee5acc743814b5ddd5e747a1b3f8, valid=YES, reason=0, block= 2430955, type= 108, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430955, 2022-07-25 idx= 3); txid: cc345570599f590e8a66d7bd3e769266a7def7b20f9a3a417e8d1a937823e5b5
2022-07-25T11:39:11Z parseTransaction(): strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, spReference: , strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c0980d293ad03, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 108
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:11Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 propertyId: 9
2022-07-25T11:39:11Z 	 amount commited: 900000000
2022-07-25T11:39:11Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:11Z recordTX(cc345570599f590e8a66d7bd3e769266a7def7b20f9a3a417e8d1a937823e5b5, valid=YES, reason=0, block= 2430955, type= 108, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430955, 2022-07-25 idx= 4); txid: 1263b3913a1b28e931d3dea5a1a76437972838a08fd8f4fd41bf7ae8a2fca6d4
2022-07-25T11:39:11Z parseTransaction(): strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, spReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, strReference: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, single_pkt: 006c0980d293ad03, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 108
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 108 (Channel Commit)
2022-07-25T11:39:11Z 	 channelAddress: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 propertyId: 9
2022-07-25T11:39:11Z 	 amount commited: 900000000
2022-07-25T11:39:11Z mastercore_handler_tx(): interp_ret: 0
2022-07-25T11:39:11Z recordTX(1263b3913a1b28e931d3dea5a1a76437972838a08fd8f4fd41bf7ae8a2fca6d4, valid=YES, reason=0, block= 2430955, type= 108, value= 0)
2022-07-25T11:39:11Z GetEncodingClass(): HAS OP RETURN!
2022-07-25T11:39:11Z ____________________________________________________________________________________________________________________________________
2022-07-25T11:39:11Z parseTransaction(block=2430955, 2022-07-25 idx= 5); txid: 9b79f2fa914be06a342c983a522431423e96243d28f98e99d6e36ea5e501f990
2022-07-25T11:39:11Z parseTransaction(): strSender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb 
2022-07-25T11:39:11Z parseTransaction(): mp_tx object: strSender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, spReference: QfNqQ37ySn5MBxjMaPVzJv8YvFJzUKmAmJ, strReference: QZwDshKJgnFzQeo3gZFMbD6vnA2oEdNhFJ, single_pkt: 0072095acdb0940180f28ba8090101, tlClass: 4 
2022-07-25T11:39:11Z interpret_TransactionType(): type: 114
2022-07-25T11:39:11Z 	------------------------------
2022-07-25T11:39:11Z 	         version: 0, class D
2022-07-25T11:39:11Z 	            type: 114 (Channel Contract Instant Trade)
2022-07-25T11:39:11Z sinterpret_Contract_Instant): inside interpret function!!!!!
2022-07-25T11:39:11Z sinterpret_Contract_Instant): function returning !!!!!
2022-07-25T11:39:11Z 	 version: 0
2022-07-25T11:39:11Z 	 messageType: 114
2022-07-25T11:39:11Z 	 contractId: 9
2022-07-25T11:39:11Z 	 amount : 90
2022-07-25T11:39:11Z 	 blockfor_expiry : 2431053
2022-07-25T11:39:11Z 	 price : 2500000000
2022-07-25T11:39:11Z 	 trading action : 1
2022-07-25T11:39:11Z 	 leverage : 1
2022-07-25T11:39:11Z logicMath_Contract_Instant: AmountToReserve: 900000000
2022-07-25T11:39:11Z logicMath_Contract_Instant: sender: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb, channel Address: QcaT3YM9apGNf3LbPeYP5m8jN4vPNQthSb
2022-07-25T11:39:11Z logicMath_Contract_Instant(): first address reserve done
2022-07-25T11:39:11Z logicMath_Contract_Instant(): second address reserve done
2022-07-25T11:39:11Z updateRecord(): now64: -1
2022-07-25T11:39:11Z updateRecord(): now64: 1
2022-07-25T11:39:11Z Instant_x_Trade: old first position: -1, new first position: -91 
2022-07-25T11:39:11Z Instant_x_Trade: old second position: 1, new second position: 91 
2022-07-25T11:39:11Z Instant_x_Trade: Status_marker0: ShortPosIncreased, Status_taker0: LongPosIncreased 
2022-07-25T11:39:11Z Instant_x_Trade: amount_forsale: 90
2022-07-25T11:39:11Z Instant_x_Trade: newPosMaker: 0
2022-07-25T11:39:11Z Instant_x_Trade: newPosTaker: 0
2022-07-25T11:39:11Z 
Check here CDEx:	 nBlockNow = 2430955	 twapBlockg = 2430952
2022-07-25T11:39:11Z 
twap_elej CDEx = 24499606741.00000000
2022-07-25T11:39:11Z 
cdextwap_ele.size() = 1
2022-07-25T11:39:11Z logicMath_Contract_Instant(): End of Logic Instant Contract Trade

**Comment by patrickdugan** [comment]
I believe this was going into asserts and tripping on them, and that the flowing tokens may not be 100% as planned since we're passing asserts as error messages now... but the performance of sync improved and not finding crashes ending on this note.
